js/gulpfile.js

```diff
@@ -185,44 +185,125 @@ gulp.task("test:common:generate", (cb) => {
     pump([gulp.src(["../scripts/Controller.ice"]), slice2js(), gulp.dest("test/Common")], cb);
 });
 
+// A rollup plug-in to resolve "net" module as a mockup file.
+
+const nodeMokups = ["net", "fs", "path"];
+function NodeMookupResolver() {
+    return {
+        name: "mookup-resolver",
+        async resolveId(source) {
+            if (nodeMokups.includes(source)) {
+                return path.join(__dirname, `test/Common/${source}.mockup.js`);
+            }
+            return null;
+        },
+    };
+}
+
+gulp.task("ice:bundle", async (cb) => {
+    let bundle = await rollup({
+        input: "src/index.js",
+        plugins: [NodeMookupResolver()],
+    });
+    return bundle.write({
+        file: "dist/ice.js",
+        format: "esm",
+    });
+});
+
+// A rollup resolver plugin to resolve "ice" module as an external file.
+function IceResolver() {
+    return {
+        name: "ice-resolver",
+        async resolveId(source) {
+            if (source == "ice") {
+                return { id: "/ice.js", external: "absolute" };
+            }
```

Comment:
We resolve `import "ice"` to an external module (not bundled) with `/ice.js` path